File: python/rgb_to_hex_conversion/rgb_to_hex.py
def rgb(r, g, b):
    rgb_arr = [r, g, b]
    rgb_arr = [0 if x < 0 else 255 if x > 255 else x for x in rgb_arr]

    r, g, b = rgb_arr

    return f"{r:02X}{g:02X}{b:02X}"

File: javascript/rgb_to_hex_conversion/rgb_to_hex.js
function rgb(r, g, b) {
  return [r, g, b].map((x) => Math.max(0, Math.min(255, x))
    .toString(16).toUpperCase().padStart(2, '0')
  ).join('')
}

// Below is my original answer and above is my refactored one

// function rgb(r, g, b) {
//   return [r, g, b].map((x) => {
//     if (x > 255) return 'FF'
//     if (x < 0) return '00'
//     return x.toString(16).toUpperCase().padStart(2, '0')
//   }).join('')
// }
module.exports = rgb;
